C/common/management_client.cpp

```diff
@@ -2059,3 +2059,67 @@ int ManagementClient::validateDatapoints(std::string dp1, std::string dp2)
 
 	return temp.compare(dp2);
 }
+
+/**
+ * Get an alert by specific key
+ *
+ * @param    key        Key to get alert
+ * @return   string     Alert
+ */
+std::string ManagementClient::getAlertByKey(const std::string& key)
+{
+	std::string response = "Status: 404 Not found";
+	try
+	{
+		std::string url = "/fledge/alert/" + urlEncode(key) ;
+		auto res = this->getHttpClient()->request("GET", url.c_str());
+		Document doc;
+		response = res->content.string();
+		doc.Parse(response.c_str());
+		if (doc.HasParseError())
+		{
+			bool httpError = (isdigit(response[0]) && isdigit(response[1]) && isdigit(response[2]) && response[3]==':');
+			m_logger->error("%s fetching alert: %s\n",
+								httpError?"HTTP error while":"Failed to parse result of",
+								response.c_str());
+		}
+	}
+	catch (const SimpleWeb::system_error &e) {
+		m_logger->error("Get alert failed %s.", e.what());
+	}
+	return response;
+}
+
+
+/**
+ * Raise an alert
+ *
+ * @param    payload        Alert payload
+ * @return   whether operation was successful
+ */
+bool ManagementClient::raiseAlert(const std::string& payload)
```

Comment:
Don't make the caller create the JSON, better to take key, urgency and message as arguments. The client is then the only place the exact payload format is required.